atom/common/api/remote_object_freer.cc

```diff
@@ -62,6 +70,10 @@ void RemoteObjectFreer::RunDestructor() {
   base::ListValue args;
   args.AppendString(context_id_);
   args.AppendInteger(object_id_);
+  args.AppendInteger(ref_mapper_[context_id_][object_id_]);
+  // Reset our local ref count in case we are in a GC race condition and will
+  // get more references in an inbound IPC message
+  ref_mapper_[context_id_][object_id_] = 0;
```

Comment:
The key should be erased otherwise memory would not be freed.

```c++
ref_mapper_[context_id_].erase(object_id_);
if (ref_mapper_[context_id_].empty())
  ref_mapper_.erase(context_id_);
```